x/stake/endblock.go

```diff
@@ -252,3 +269,46 @@ func handleMatureUnbondingDelegations(k keeper.Keeper, ctx sdk.Context) ([]types
 
 	return completed, events
 }
+
+const (
+	maxProcessedRefundCount = 10
+)
+
+func handleRefundStake(ctx sdk.Context, k keeper.Keeper) sdk.Events {
+	iterator := k.IteratorAllDelegations(ctx)
+	defer iterator.Close()
+	var refundEvents sdk.Events
+	count := 0
+
+	bscValidators := k.GetAllValidators(ctx)
+	bscValidatorsMap := make(map[string]types.Validator, len(bscValidators))
+	for _, validator := range bscValidators {
+		bscValidatorsMap[validator.OperatorAddr.String()] = validator
+	}
+	boundDenom := k.BondDenom(ctx)
+
+	for ; iterator.Valid(); iterator.Next() {
+		delegation := types.MustUnmarshalDelegation(k.CDC(), iterator.Key(), iterator.Value())
+		validator := bscValidatorsMap[delegation.ValidatorAddr.String()]
+		amount := validator.TokensFromShares(delegation.GetShares()).RawInt()
```

Comment:
In our current implementation, `delegation.GetShares()` is exactly the same with the staked BNB amount, so I think no need to convert and `GetAllValidators`

Reply:
fixed